neo4jgraph.py

```
@@ -0,0 +1,211 @@
+from neo4j import GraphDatabase
+from neo4j.exceptions import ServiceUnavailable
+import logging
+
+
+class Neo4jGraph:
+    def __init__(self, uri, user, password):
+        self.driver = GraphDatabase.driver(uri, auth=(user, password), encrypted=False)
+
+    def close(self):
+        self.driver.close()
+
+    def run_query(self, query):
+        # Function to allow run any cypher query
+        with self.driver.session() as session:
+            result = session.write_transaction(self._run_query, query)
+            for row in result:
+                print(row)
+
+    @staticmethod
+    def _run_query(tx, query):
+        return tx.run(query)
+
+    def generate_dc(self, dc_id):
+        with self.driver.session() as session:
+            result = session.write_transaction(self._generate_dc, dc_id)
+            for dc_id in result:
+                print(f"Generated dataset collection {dc_id}")
+
+    @staticmethod
+    def _generate_dc(tx, dc_id):
+        # Generate a dataset collection node
+        query = (
+            f'CREATE (dc:dataset_collection {{dc_id: {dc_id}, description: "Dataset collection number {dc_id}"}}) '
+            f'RETURN dc'
+        )
+
+        result = tx.run(query)
+        try:
+            return [row["dc"]["dc_id"] for row in result]
+
+        # Capture any errors along with the query and data for traceability
+        except ServiceUnavailable as exception:
+            logging.error("{query} raised an error: \n {exception}".format(
+                query=query, exception=exception))
+            raise
```

I think this should raise an Error(your logging message) and the generate_dc should handle the error (probably propagate it as well, and the end user of generate_dc will decide what to do with it)